src/components/data/DataTable.js

```diff
@@ -0,0 +1,151 @@
+import { downcastRef } from '../../util/typing';
+import { useArrowKeyNavigation } from '../../hooks/keyboard-navigation';
+import { useElementRef } from '../../hooks/use-element-ref';
+
+import { SpinnerSpokesIcon } from '../icons';
+import Table from './Table';
+import TableHead from './TableHead';
+import TableBody from './TableBody';
+import TableRow from './TableRow';
+import TableCell from './TableCell';
+/**
+ * @typedef {import('../../types').CompositeProps} CommonProps
+ * @typedef {Omit<import('preact').JSX.HTMLAttributes<HTMLElement>, 'size'|'rows'|'role'|'loading'>} HTMLAttributes
+ * @typedef {import('./TableContext').TableInfo} TableInfo
+ * @typedef {import('preact').ComponentChildren} Children
+ *
+ * @typedef Column
```

Comment:
In these newer Table components, terminology is tighter:

* `columns` vs. legacy `tableHeaders`
* `rows` vs. `items`
* An `item` in new components is an individual cell, not a row